xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp

```diff
@@ -1221,11 +1225,7 @@ void CAESinkAUDIOTRACK::UpdateAvailablePCMCapabilities()
 
   int encoding = CJNIAudioFormat::ENCODING_PCM_16BIT;
   m_sinkSupportsFloat = VerifySinkConfiguration(native_sampleRate, CJNIAudioFormat::CHANNEL_OUT_STEREO, CJNIAudioFormat::ENCODING_PCM_FLOAT);
-  // Only try for Android 7 or later - there are a lot of old devices that open successfully
-  // but won't work correctly under the hood (famouse example: old FireTV)
-  // As even newish devices like Android Chromecast don't do it properly - just disable it ... and use 16 bit Integer
-  //if (CJNIAudioManager::GetSDKVersion() > 23)
-  //  m_sinkSupportsMultiChannelFloat = VerifySinkConfiguration(native_sampleRate, CJNIAudioFormat::CHANNEL_OUT_7POINT1_SURROUND, CJNIAudioFormat::ENCODING_PCM_FLOAT);
+  m_sinkSupportsMultiChannelFloat = VerifySinkConfiguration(native_sampleRate, CJNIAudioFormat::CHANNEL_OUT_7POINT1_SURROUND, CJNIAudioFormat::ENCODING_PCM_FLOAT);
```

Comment:
Not should continue test only if android API supports ?

`if (CJNIAudioManager::GetSDKVersion() > 23)`

Reply:
I thought about this, but:  https://developer.android.com/reference/android/media/AudioFormat#ENCODING_PCM_FLOAT is available since v21. So I would enumerate it always? As v21 is Android 5 - do you think we run on Android 4 somewhere?

Reply:
The v23 was the old workaround already ... as some devices should not use it. I can get it back in - your choice.

Reply:
As now is the advanced setting, is not need the workaround here but maintain the formal Android version check only.

Reply:
Yes - then I would add v21 - thanks.